src/libgetenv/src/getenv.cpp

```diff
@@ -322,7 +326,11 @@ void applyOptions()
 	if ((k = ksLookupByName(elektraConfig, "/env/option/clearenv", 0)))
 	{
 		LOG << "clearing the environment" << endl;
+#ifdef HAVE_CLEARENV
 		clearenv();
+#else
+		environ = NULL;
```

Comment:
this does not really remove anything, the environment would still be accessible for everyone who has saved the pointer.
Isn't there a mac os x substitute for clearenv? (but I am afraid it is not)

A loop with unsetenv() shouldn't be too hard to write though (src/libgetenv/examples/getenv already does similar stuff)